layers/timeline_semaphore.c

```diff
@@ -2067,6 +2067,187 @@ static void timeline_GetPhysicalDeviceExternalSemaphoreProperties(
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_INTEGER_FUNCTIONS_2_FEATURES_INTEL:
+        return sizeof(VkPhysicalDeviceShaderIntegerFunctions2FeaturesINTEL);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_SM_BUILTINS_FEATURES_NV:
+        return sizeof(VkPhysicalDeviceShaderSMBuiltinsFeaturesNV);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_SUBGROUP_EXTENDED_TYPES_FEATURES:
+        return sizeof(VkPhysicalDeviceShaderSubgroupExtendedTypesFeatures);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADING_RATE_IMAGE_FEATURES_NV:
+        return sizeof(VkPhysicalDeviceShadingRateImageFeaturesNV);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_SIZE_CONTROL_FEATURES_EXT:
+        return sizeof(VkPhysicalDeviceSubgroupSizeControlFeaturesEXT);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TEXEL_BUFFER_ALIGNMENT_FEATURES_EXT:
+        return sizeof(VkPhysicalDeviceTexelBufferAlignmentFeaturesEXT);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TEXTURE_COMPRESSION_ASTC_HDR_FEATURES_EXT:
+        return sizeof(VkPhysicalDeviceTextureCompressionASTCHDRFeaturesEXT);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TIMELINE_SEMAPHORE_FEATURES:
+        return sizeof(VkPhysicalDeviceTimelineSemaphoreFeatures);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TRANSFORM_FEEDBACK_FEATURES_EXT:
+        return sizeof(VkPhysicalDeviceTransformFeedbackFeaturesEXT);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_UNIFORM_BUFFER_STANDARD_LAYOUT_FEATURES:
+        return sizeof(VkPhysicalDeviceUniformBufferStandardLayoutFeatures);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VARIABLE_POINTERS_FEATURES:
+        return sizeof(VkPhysicalDeviceVariablePointersFeatures);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VERTEX_ATTRIBUTE_DIVISOR_FEATURES_EXT:
+        return sizeof(VkPhysicalDeviceVertexAttributeDivisorFeaturesEXT);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_1_FEATURES:
+        return sizeof(VkPhysicalDeviceVulkan11Features);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_2_FEATURES:
+        return sizeof(VkPhysicalDeviceVulkan12Features);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_MEMORY_MODEL_FEATURES:
+        return sizeof(VkPhysicalDeviceVulkanMemoryModelFeatures);
+    case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_YCBCR_IMAGE_ARRAYS_FEATURES_EXT:
+        return sizeof(VkPhysicalDeviceYcbcrImageArraysFeaturesEXT);
+    default:
+        unreachable("Unknown structure for VkDeviceCreateInfo::pNext");
+    }
+    return 0;
+}
+
+static void* clone_device_create_info_pnext(
+    const VkAllocationCallbacks *pAllocator,
+    const void *pNext)
+{
+    VkBaseOutStructure *head = NULL;
+    VkBaseOutStructure *tail = NULL;
+
+    vk_foreach_struct_const(item, pNext) {
+        size_t item_size = vk_device_create_info_type_size(item);
+        struct VkBaseOutStructure *new_item =
+            vk_alloc(pAllocator, item_size, 8, VK_SYSTEM_ALLOCATION_SCOPE_DEVICE);
+
+        if (new_item == NULL)
+            return NULL;
```

Comment:
Probably want to call `free_device_create_info_pnext` here.

Reply:
Right, thanks for catching this.